lib/rules/no-new-func.js

```diff
@@ -38,14 +38,26 @@ module.exports = {
                     variable.references.forEach(ref => {
                         const node = ref.identifier;
                         const { parent } = node;
+                        let evalNode;
 
-                        if (
-                            parent &&
-                            (parent.type === "NewExpression" || parent.type === "CallExpression") &&
-                            node === parent.callee
-                        ) {
+                        if (parent) {
+                            if (node === parent.callee && (
+                                parent.type === "NewExpression" ||
+                                parent.type === "CallExpression"
+                            )) {
+                                evalNode = parent;
+                            } else if (parent.type === "MemberExpression" && node === parent.object) {
+                                const maybeCallee = parent.parent.type === "ChainExpression" ? parent.parent : parent;
+
+                                if (maybeCallee.parent.type === "CallExpression" && maybeCallee.parent.callee === maybeCallee) {
+                                    evalNode = parent.parent;
```

Comment:
```suggestion
                                    evalNode = maybeCallee.parent;
```

This correctly reports `(Function?.call)(...)` as disallowed code, but flags `Function?.call` instead of ``(Function?.call)(...)``.

We can add the following test case to confirm that the rule reports this code and to assert that the right node is reported:

```js
{
    code: "var a = (Function?.call)(null, \"b\", \"c\", \"return b+c\");",
    parserOptions: { ecmaVersion: 2021 },
    errors: [{
        messageId: "noFunctionConstructor",
        type: "CallExpression"
    }]
}
```